services/memory_storage/memory_storage.cpp

```diff
@@ -256,19 +274,82 @@ namespace services {
                   session.data());
             return;
         }
+        auto collection_names = logical_plan->collection_dependencies();
+        context_storage_t collection_dependencies;
+        for (const auto& name : collection_names) {
+            if (!check_collection_(session, name)) {
+                trace(log_,
+                      "memory_storage_t:execute_plan_ collection not found {}, sesion: {}",
+                      name.to_string(),
+                      session.data());
+                return;
+            }
+            collection_dependencies.emplace(name, collections_.at(name).context);
+        }
         sessions_.emplace(session, session_t{logical_plan, current_message()->sender(), 1});
-        actor_zeta::send(collections_.at(logical_plan->collection_full()),
+        auto plan =
+            collection::planner::create_plan(collection_dependencies, logical_plan, components::ql::limit_t::unlimit());
+
+        if (!plan) {
+            execute_plan_finish_(
+                session,
+                make_cursor(default_resource(), error_code_t::create_phisical_plan_error, "invalid query plan"));
+            return;
+        }
+
+        std::stack<collection::operators::operator_ptr> look_up;
+        std::stack<collection::operators::operator_ptr> sub_plans;
+        look_up.push(plan);
+        while (!look_up.empty()) {
+            collection::operators::operator_ptr check_op = look_up.top();
+            while (check_op->right() == nullptr) {
+                check_op = check_op->left();
+                if (check_op == nullptr) {
+                    break;
+                }
+            }
+            sub_plans.push(look_up.top());
+            look_up.pop();
+            if (check_op != nullptr) {
+                look_up.push(check_op->right());
+                look_up.push(check_op->left());
+            }
+        }
+
+        auto current_plan =
```

Comment:
We set as first operator  leftmost bottom in the tree plan. I can't get how all this stack will be executed, could you plz point me?

Reply:
It is a start of a sequence. Send sub_plan to execute, then get callback, execute next and so on. I will add a comment for it.